sklearn/linear_model/tests/test_logistic.py

```diff
@@ -2065,6 +2076,47 @@ def test_liblinear_not_stuck():
         clf.fit(X_prep, y)
 
 
+def test_lr_cv_scores_differ_when_sample_weight_is_requested():
+    rng = np.random.RandomState(10)
+    X, y = make_classification(n_samples=10, random_state=rng)
+    sample_weight = np.ones(len(y))
+    sample_weight[: len(y) // 2] = 2
+    kwargs = {"sample_weight": sample_weight}
+
+    with config_context(enable_metadata_routing=True):
+        scorer1 = get_scorer("accuracy")
+        lr_cv1 = LogisticRegressionCV(scoring=scorer1)
+        lr_cv1.fit(X, y, **kwargs)
+
+        scorer2 = get_scorer("accuracy")
+        scorer2.set_score_request(sample_weight=True)
+        lr_cv2 = LogisticRegressionCV(scoring=scorer2)
+        lr_cv2.fit(X, y, **kwargs)
+
+    assert pytest.approx(lr_cv1.scores_[1]) != lr_cv2.scores_[1]
+
+
+def test_lr_cv_scores_without_enabling_metadata_routing():
+    rng = np.random.RandomState(10)
+    X, y = make_classification(n_samples=10, random_state=rng)
+    sample_weight = np.ones(len(y))
+    sample_weight[: len(y) // 2] = 2
+    kwargs = {"sample_weight": sample_weight}
+
+    with config_context(enable_metadata_routing=False):
+        scorer1 = get_scorer("accuracy")
+        lr_cv1 = LogisticRegressionCV(scoring=scorer1)
+        lr_cv1.fit(X, y, **kwargs)
+
+    with config_context(enable_metadata_routing=True):
+        scorer2 = get_scorer("accuracy")
+        scorer2.set_score_request(sample_weight=True)
+        lr_cv2 = LogisticRegressionCV(scoring=scorer2)
+        lr_cv2.fit(X, y, **kwargs)
+
+    assert_almost_equal(lr_cv1.scores_[1], lr_cv2.scores_[1])
+
```

Comment:
We should also test `lr_cv1.score(X, y)` and `lr_cv2.score(X,  y)` to check that they provide the same score.

Reply:
These tests are basically testing for the routing in the fit method. I am not sure if the score method would return exactly the same results.

Reply:
I don't think this test would apply to the score method because the score method by default does not pass sample_weight ahead currently. This test is basically checking that when we request sample_weight through metadata routing in fit as compared to when we don't use metadata routing but still set sample_weight the results are the same.

Reply:
Since we add `**score_params`, the `score` method will request the same meta data through the `scoring` object pass at `__init__` if I am not wrong.

@adrinjalali can you confirm that I don't say anything stupid here?

Reply:
The first case in this test would not pass sample_weight to the scorer inside the score method since enable_metadata_routing=False.

Reply:
Wait I might make a mistake, I meant `lr_cv1.score(X, y, **kwargs)` == `lr_cv2.score(X, y, **kwargs)`.

We currently have `sample_weight` in the `score` function of `LogisticRegressionCV` so with or without metadata routing, `sample_weight` should be dispatched to the internal `scoring`.

Reply:
We currently were not passing sample_weight to the internal scoring.

Reply:
I think I understand now. I was a bit confused because I thought we were not passing sample_weight to the internal scoring.

Reply:
It might have actually be a bug solved in the last release :)

Reply:
Thanks let me fix this. I think we might need to make some adjustment in the code as well.

Reply:
@adrinjalali I think we might need to add sample_weight in the default case as well.

Reply:
So both of you are right and wrong at the same time. Or "yes and no" as Sir Humphrey would say in Yes Minister!

Right now we do pass `sample_weight` to the underlying scorer in `main`, so we should do the same (needs to be fixed in the PR). That means in this particular case, the two `score` methods should return the same value.

However, we do not pass other `kwargs` to the underlying scorer in `main` now, so if a scorer requests those, then there will be a difference between the two cases.